sessionize/app/src/main/java/co/touchlab/sessionize/NotificationsApiImpl.kt

```diff
@@ -14,14 +14,70 @@ import android.os.Build
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationManagerCompat
 import app.sessionize.touchlab.lib.R
+import co.touchlab.droidcon.db.MySessions
 import co.touchlab.sessionize.api.NotificationsApi
+import co.touchlab.sessionize.api.notificationFeedbackTag
+import co.touchlab.sessionize.api.notificationReminderTag
 import co.touchlab.sessionize.platform.AndroidAppContext
+import co.touchlab.sessionize.platform.AndroidAppContext.backgroundTask
+import co.touchlab.sessionize.platform.currentTimeMillis
+import co.touchlab.sessionize.platform.feedbackEnabled
+import co.touchlab.sessionize.platform.notificationsEnabled
+import co.touchlab.sessionize.platform.reminderNotificationsEnabled
+import co.touchlab.sessionize.platform.setNotificationsEnabled
 
 
 class NotificationsApiImpl : NotificationsApi {
 
     private val notificationPublisher: BroadcastReceiver = NotificationPublisher()
 
+    override fun createReminderNotificationsForSessions(sessions:List<MySessions>){
+        if(reminderNotificationsEnabled() && notificationsEnabled()) {
```

Comment:
These functions are really business logic which should happen before calling out to the platform. They don't have any platform code and are identical between android and iOS. This class should really be as simple as possible and only handle the translation of the common code's desires "create a notification" into android specific calls.

Reply:
Moved in latest commit.